network/packethandlers.c

```diff
@@ -2,4 +2,598 @@
+
+    llnet_packet_free(rawPacket);
+    return packet;
+}
+
+PacketTLV_t* unpackConfigResponse(IntermediateTLV_t* rawPacket)
+{
+    PacketTLV_t* packet = createBasePacket(rawPacket);
+    if (packet == NULL)
+    {
+        //Free the raw packet
+        llnet_packet_free(rawPacket);
+        return NULL;
+    }
+
+    errno = 0;
+    PTLVData_CONFIG_RESPONSE_t* unpacked = calloc(1,sizeof(PTLVData_CONFIG_RESPONSE_t));
+    //Allocation failed
+    if(unpacked == NULL && errno != 0)
+    {
+        //packet allocation succeeded, so free that memory
+        free(packet);
+        //Free the raw packet
+        llnet_packet_free(rawPacket);
+        return NULL;
+    }
+    packet->type = rawPacket->type;
+    packet->timestamp = rawPacket->timestamp;
+    packet->length = rawPacket->length;
+    packet->data = (PTLVData_Base_t*)unpacked;
+    unpacked->pairs = parseKVList((char*)rawPacket->data, rawPacket->length);
+
+    llnet_packet_free(rawPacket);
+    return packet;
+}
+
+PacketTLV_t* unpackConfigUpdate(IntermediateTLV_t* rawPacket)
+{
+    PacketTLV_t* packet = createBasePacket(rawPacket);
+    if (packet == NULL)
+    {
+        //Free the raw packet
+        llnet_packet_free(rawPacket);
+        return NULL;
+    }
+
+    errno = 0;
+    PTLVData_CONFIG_UPDATE_t* unpacked = calloc(1,sizeof(PTLVData_CONFIG_UPDATE_t));
+    //Allocation failed
+    if(unpacked == NULL && errno != 0)
+    {
+        //packet allocation succeeded, so free that memory
+        free(packet);
+        //Free the raw packet
+        llnet_packet_free(rawPacket);
+        return NULL;
+    }
+    packet->length = rawPacket->length;
+    packet->timestamp = rawPacket->timestamp;
+    packet->type = rawPacket->type;
+    packet->data = (PTLVData_Base_t*)unpacked;
+
+    //Note that this may be NULL if there's a failure
+    unpacked->new_pairs = parseKVList((char*)rawPacket->data, rawPacket->length);
+
+    llnet_packet_free(rawPacket);
+    return packet;
+}
+
+PacketTLV_t* unpackUserData(IntermediateTLV_t* rawPacket)
+{
+    PacketTLV_t* packet = createBasePacket(rawPacket);
+    if (packet == NULL)
+    {
+        //Free the raw packet
+        llnet_packet_free(rawPacket);
+        return NULL;
+    }
+
+    errno = 0;
+    PTLVData_USER_DATA_t* unpacked = calloc(1,sizeof(PTLVData_USER_DATA_t));
+    //Allocation failed
+    if(unpacked == NULL && errno != 0)
+    {
+        //packet allocation succeeded, so free that memory
+        free(packet);
+        //Free the raw packet
+        llnet_packet_free(rawPacket);
+        return NULL;
+    }
+
+    packet->data = (PTLVData_Base_t*)unpacked;
+
+    unpacked->left_stick_x = rawPacket->data[0];
+    unpacked->left_stick_y = rawPacket->data[1];
+    unpacked->right_stick_x = rawPacket->data[2];
+    unpacked->right_stick_y = rawPacket->data[3];
+    unpacked->button_a = (rawPacket->data[4] & 0x80) != 0;
+    unpacked->button_b = (rawPacket->data[4] & 0x40) != 0;
```

Comment:
what about buttons 3 and 4?

Reply:
Only A and B are implemented in the standard, but it's right next to a reserved block.  I'll expand it in the next update for the protocol